irohad/main/impl/on_demand_ordering_init.cpp

```diff
@@ -171,38 +172,62 @@ namespace iroha {
             proposal_factory,
         std::shared_ptr<ametsuchi::TxPresenceCache> tx_cache,
         consensus::Round initial_round) {
+      auto delay = [reject_counter = static_cast<uint64_t>(0),
+                    local_counter =
+                        static_cast<uint64_t>(0)](const auto &commit) mutable {
+        using iroha::synchronizer::SynchronizationOutcomeType;
+        if (commit.sync_outcome == SynchronizationOutcomeType::kReject
+            or commit.sync_outcome == SynchronizationOutcomeType::kNothing) {
+          if (local_counter++ == 2) {
+            local_counter = 0;
+            if (reject_counter < std::numeric_limits<uint64_t>::max()) {
+              reject_counter++;
+            }
+          }
+        } else {
+          reject_counter = 0;
+        }
+        return std::chrono::seconds(reject_counter);
```

Comment:
What is the point to increase delay each reject round? I think better to increase rejects only after f malicious rounds because first f rounds could be conducted by malicious os.